src/cross_section/src/cross_section.cpp

```diff
@@ -304,6 +304,129 @@ std::shared_ptr<const PathImpl> CrossSection::GetPaths() const {
   return paths_;
 }
 
+/**
+ * Constructs a CrossSection from a vector of polygons.
+ * No checking is done, so don't have self-intersecting paths
+ * Polygon contours should be wound CCW.
+ * Polygon holes should be wound CW.
+ *
+ * @param polygons A vector of a vector of path points.
+ */
+CrossSection CrossSection::CreateFromPolygonsUnchecked(
+    std::vector<std::vector<glm::vec2>> polygons) {
+  if (polygons.size() < 1) {
+    return CrossSection();
+  }
+  for (int i = 0; i < polygons.size(); i++) {
+    if (polygons[i].size() < 3) {
+      return CrossSection();
+    }
+  }
+
+  auto ps = C2::PathsD(polygons.size());
+
+  for (int i = 0; i < ps.size(); i++) {
+    auto path = polygons[i];
+    auto p = C2::PathD(polygons[i].size());
+    for (int j = 0; j < p.size(); j++) {
+      auto v = path[j];
+      p[j] = C2::PointD(v.x, v.y);
+    }
+    ps[i] = p;
+  }
+  return CrossSection(shared_paths(ps));
+}
+
+/**
+ * Constructs a rounded rectangle from given size and radius.
+ *
+ * @param size The X, and Y dimensions of the square.
+ * @param radius Radius of the circle. Must be positive.
+ * @param circularSegments Number of segments along its diameter. Default is
+ * calculated by the static Quality defaults according to the radius.
+ * @param center Set to true to shift the center to the origin.
+ */
+CrossSection CrossSection::RoundedRectangle(const glm::vec2 size, float radius,
```

Comment:
This is quite a bit of extra code to maintain for a pretty narrow use case. Also, did you intend to include tests?

Reply:
It's really not that narrow of a use case. It was one of the most asked questions with OpenSCAD (which did not have a good way to do this). Note I would not need to ask if you actually provided a way to make this from outside Manifold. Currently I lose an order of magnitude due to artificial overhead.

Reply:
@elalish As to tests, for the parts you think you might want to accept, I'll gladly add tests.
I frankly am skeptical that you'll accept any of it.
But I had done the work, I'm simply offering it for your potential use.

Reply:
Accepting external code is not quite like getting mana from heaven. That you say this makes me think you have not yet accepted external code into one of your projects?